packages/lit-html/src/hydrate.ts

```diff
@@ -255,22 +255,17 @@ const openChildPart = (
     // }
   } else if (isTemplateResult(value)) {
     // Check for a template result digest
-    const markerWithDigest = `lit-part ${digestForTemplateResult(
-      value as TemplateResult
-    )}`;
+    const markerWithDigest = `lit-part ${digestForTemplateResult(value)}`;
     if (marker.data === markerWithDigest) {
-      const template = ChildPart.prototype._$getTemplate(
-        (value as TemplateResult).strings,
-        value as TemplateResult
-      );
+      const template = ChildPart.prototype._$getTemplate(value);
```

Comment:
btw, this isn't doing much on the prototype. I think we should move it to static like with `Template.createElement`.